server/src/main/java/org/opensearch/plugins/NoOpResourceAccessControlPlugin.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import org.opensearch.accesscontrol.resources.EntityType;
+import org.opensearch.accesscontrol.resources.ResourceSharing;
+import org.opensearch.accesscontrol.resources.ShareWith;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This plugin class defines a no-op implementation of Resource Plugin.
+ *
+ * @opensearch.experimental
+ */
+public class NoOpResourceAccessControlPlugin implements ResourceAccessControlPlugin {
+
+    /**
+     *
+     * @return an empty map of resource names accessible by this user.
+     */
+    @Override
+    public Map<String, List<String>> listAccessibleResources() {
+        return Map.of();
+    }
+
+    /**
+     * Returns an empty list since security plugin is not defined.
+     * This method alone doesn't determine permissions.
+     *
+     * @return empty list
+     */
+    @Override
+    public List<String> listAccessibleResourcesForPlugin(String systemIndexName) {
```

Comment:
I think any instance of systemIndex should be renamed to resourceIndex. A `resourceIndex` is a system index, but it would specifically be for plugins that want to have this resource-sharing ability.